includes/class-wc-payments.php

```diff
@@ -1813,6 +1813,9 @@ public static function init_woopay() {
 	 * @return void
 	 */
 	public static function load_stripe_bnpl_site_messaging() {
+		if ( ! self::get_gateway()->is_enabled() ) {
```

Comment:
(minor) For this, I was wondering if we should instead adding this check a few lines above, before the hooks are registered, with the other check: https://github.com/Automattic/woocommerce-payments/blob/525d198c5e99bf9bc85b5f5d7d2f0bd916eba12e/includes/class-wc-payments.php#L621

What do you think?

Reply:
Good idea! I implemented it in https://github.com/Automattic/woocommerce-payments/pull/9467/commits/b6c7d059875f9d1029d97270950d2d7d939f2ce8